spec/cypress/support/step_definitions/factory-bot.js

```diff
@@ -18,12 +18,36 @@ expect(parseArgs('a "a b", b "b" and c "d"')).to.deep.equal({
   c: "d",
 });
 
+Given("{word} was created", (factory) => {
+  cy.appFactories([["create", factory]]);
+});
+
+Given("{word} was created with {}", (factory, args) => {
+  cy.appFactories([["create", factory, parseArgs(args)]]);
+});
+
 Given("{word} was created as {string} with {}", (factory, traits, args) => {
   cy.appFactories([
     ["create", factory, ...traits.split(", "), parseArgs(args)],
   ]);
 });
 
-Given("{word} was created with {}", (factory, args) => {
-  cy.appFactories([["create", factory, parseArgs(args)]]);
-});
+const login = (traits, args) => {
+  const factoryArgs = ["create", "user", ...traits.split(", ")];
+
+  if (args) {
+    factoryArgs.push(parseArgs(args));
+  }
+
+  cy.appFactories([factoryArgs])
+    .as("userData")
+    .then(([user]) => {
+      cy.visit("/users/sign_in");
+      cy.get("[name*=email]").type(`${user.email}{enter}`);
+    });
+};
+
+Given("I am logged in as {string}", (traits) => login(traits));
+Given("I am logged in as {string} with {}", (traits, args) =>
+  login(traits, args)
+);
```

Comment:
Are there any places where we could use it already?

Reply:
I don't think so - it's more for self-referential stuff while most of the stuff on this repo is about modifying other peoples accounts not your own